x-pack/plugins/security_solution/public/management/cypress/e2e/endpoint/artifacts.cy.ts

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { recurse } from 'cypress-recurse';
+import { APP_ENDPOINTS_PATH } from '../../../../../common/constants';
+import { getArtifactsListTestsData } from '../../fixtures/artifacts_page';
+import { API_HEADER, removeAllArtifacts } from '../../tasks/artifacts';
+import { loadEndpointDataForEventFiltersIfNeeded } from '../../tasks/load_endpoint_data';
+import { login } from '../../tasks/login';
+import { performUserActions } from '../../tasks/perform_user_actions';
+
+const yieldEndpointPolicyRevision = (): Cypress.Chainable<number> =>
+  cy
+    .request({
+      method: 'GET',
+      url: '/api/fleet/package_policies?kuery=ingest-package-policies.package.name%3A%20endpoint',
+      headers: API_HEADER,
+    })
+    .then(({ body }) => {
+      return body.items?.[0]?.revision ?? -1;
+    });
+
+const yieldInstalledEndpointRevision = (): Cypress.Chainable<number> =>
+  cy
+    .request({ method: 'GET', url: '/api/endpoint/metadata', headers: API_HEADER })
+    .then(({ body }) => {
+      expect(body.data.length).is.lte(1); // during update it can be temporary zero
+      return body.data?.[0]?.policy_info?.endpoint?.revision ?? -1;
+    });
+
+const parseRevNumber = (revString: string) => Number(revString.match(/\d+/)?.[0]);
+
+describe('Artifact pages', () => {
+  before(() => {
+    login();
+    loadEndpointDataForEventFiltersIfNeeded();
+    removeAllArtifacts();
+
+    // wait for ManifestManager to pick up artifact changes that happened either here
+    // or in a previous test suite `after`
+    cy.wait(6000); //  packagerTaskInterval + 1s
+
+    yieldEndpointPolicyRevision().then((actualEndpointPolicyRevision) => {
+      const hasReachedActualRevision = (revision: number) =>
+        revision === actualEndpointPolicyRevision;
+
+      // need to wait until revision is bumped to ensure test success
+      recurse(yieldInstalledEndpointRevision, hasReachedActualRevision, { delay: 1500 });
+    });
+  });
+
+  beforeEach(() => {
+    login();
+  });
+
+  after(() => {
+    removeAllArtifacts();
+  });
+
+  for (const testData of getArtifactsListTestsData()) {
+    describe(`${testData.title}`, () => {
+      it(`should update Endpoint Policy on Endpoint when adding ${testData.artifactName}`, () => {
+        cy.visit(APP_ENDPOINTS_PATH);
+
+        cy.getBySel('policyListRevNo')
+          .eq(0)
+          .invoke('text')
+          .then(parseRevNumber)
+          .then((initialRevisionNumber) => {
+            cy.visit(`/app/security/administration/${testData.urlPath}`);
+
+            cy.getBySel(`${testData.pagePrefix}-emptyState-addButton`).click();
+            performUserActions(testData.create.formActions);
```

Comment:
I've also found out that the event filter select action seems to be unstable, so we may want to consider using `cypress-recurse` to verify that the selected value matches the expected input value
https://github.com/elastic/kibana/compare/main...patrykkopycinski:test/e2e-endpoint-tests-for-policy-updates2?expand=1#diff-4a693ebd1c90fa276ee880d76eee7cf80ae068381823d69b9637b2f3acea2816R25-R49

Reply:
yes, I also met this issue. it's because the dropdowns in the event filters form work differently as other dropdowns: they work only if you keep the browser in the front, so unfortunately `recurse` won't help I think

of course, running the tests with headless browser works perfectly